File: qa-artifacts/startup-handoff-native-build-20260910/independent-review/verify.mjs
import fs from 'node:fs';
import crypto from 'node:crypto';
import cp from 'node:child_process';
import assert from 'node:assert/strict';
import path from 'node:path';
import {fileURLToPath} from 'node:url';
const q=path.resolve(path.dirname(fileURLToPath(import.meta.url)),'..');
const root=path.resolve(q,'../..'),s=path.join(root,'tools/symphony/source-import-candidate');
const j=n=>JSON.parse(fs.readFileSync(path.join(q,n))),h=b=>crypto.createHash('sha256').update(b).digest('hex'),hf=f=>h(fs.readFileSync(f));
const m=j('source-manifest.json'),r=j('build-receipt.json');
const git=(...args)=>cp.execFileSync('git',['-C',s,...args],{maxBuffer:20000000});
assert.equal(git('rev-parse','HEAD').toString().trim(),'2ea834a1e2385c34b8f516854b5317c6a08a0607');
assert.equal(m.files.length,148);
for(const f of m.files){assert.equal(hf(path.join(s,f.path)),f.sha256,f.path);assert.equal(h(git('show',r.sourceCommit+':'+f.path)),f.sha256,f.path);}
assert.equal(h(m.files.map(f=>f.path+'\0'+f.sha256+'\n').join('')),r.sourceTreeSHA256);
const files=root=>fs.readdirSync(root,{recursive:true}).filter(f=>fs.statSync(path.join(root,f)).isFile()).sort();
for(const [root,map,count] of [[r.artifact,r.bundleFiles,3],[r.embeddedFrontendPath,r.embeddedFiles,7]]){assert.equal(Object.keys(map).length,count);assert.deepEqual(files(root),Object.keys(map).sort());for(const [f,v] of Object.entries(map))assert.equal(hf(path.join(root,f)),v);}
assert.equal(hf(r.binary),'37c8751c631631477579ce848c90620d375495181394471a3214fe07be261c86');
assert.deepEqual(j('source-before.json'),j('source-after.json'));
const p=j('preserved-after.json');assert.equal(Object.keys(p).length,525);assert.deepEqual(j('preserved-before.json'),p);for(const [f,v] of Object.entries(p))assert.equal(hf(f),v,f);
const deps=fs.readFileSync(path.join(q,'native-asset-dependencies.d'),'utf8').replaceAll('\\ ',' ');for(const f of Object.keys(r.embeddedFiles))assert(deps.includes(r.embeddedFrontendPath+'/'+f),f);
assert.equal(hf(path.join(s,'qa-artifacts/cross-platform-shell-20260907/candidate4-runtime-r2/web/desktop-host.mjs')),r.embeddedFiles['desktop-host.mjs']);
assert(!fs.existsSync(path.join(q,'profile-reserved-not-launched')));
console.log('PASS: 148 source/blob, 3 bundle, 7 frontend, 525 preservation, bridge, dependency mapping, profile absent');
console.log('tree',git('rev-parse','HEAD^{tree}').toString().trim());
console.log('status',git('status','--porcelain').toString());
console.log('override',JSON.stringify(r.effectiveConfigOverride));

File: qa-artifacts/sent-draft-native-build-20260910/independent-review/verify.py
from pathlib import Path
import json,hashlib,subprocess,os,plistlib
r=Path.cwd();ev=r/'qa-artifacts/sent-draft-native-build-20260910';out=ev/'independent-review';repo=r/'tools/symphony/source-import-candidate'
def j(p):return json.loads(p.read_text())
def sha(b):return hashlib.sha256(b).hexdigest()
def git(*a):return subprocess.check_output(['git','-C',str(repo),*a],env={**os.environ,'GIT_OPTIONAL_LOCKS':'0'})
c='c84cc2d07fffc1050c2be52ef31b62fbd3c1181a';tree='df3dd55c10992ee653bfb9fb82a9bd373151f7b0';aggregate='ac15ded7187b8c85fbbfb600bfc196e4e55234f2d2c8132e41fbcb0584acacc8'
m=j(ev/'source-manifest.json');before=j(ev/'source-before.json');assert before==j(ev/'source-after.json');assert before['commit']==c and before['gitTree']==tree and before['treeSHA256']==aggregate
assert m==j(r/'qa-artifacts/sent-draft-source-export-20260910/source-manifest.json');assert m['fileCount']==len(m['files'])==148
assert before['files']=={f['path']:f['sha256'] for f in m['files']};assert set(git('ls-tree','-r','--name-only',c).decode().splitlines())==set(before['files'])
for f in m['files']:
 b=git('show',c+':'+f['path']);assert sha(b)==f['sha256'] and len(b)==f['bytes'];assert b==(repo/f['path']).read_bytes()
assert sha(''.join(f['path']+'\0'+f['sha256']+'\n' for f in m['files']).encode())==aggregate;assert sum(f['bytes'] for f in m['files'])==m['totalBytes']
assert git('rev-parse','HEAD').decode().strip()==c and git('rev-parse',c+'^{tree}').decode().strip()==tree
status=git('status','--porcelain').decode().splitlines();assert status==j(r/'tools/symphony/source-stage/GIT_IMPORT_CANDIDATE.json')['preservedUntrackedPaths']
br=j(ev/'build-receipt.json');assert br['sourceCommit']==c and br['sourceTreeSHA256']==aggregate
app=Path(br['artifact']);assets=j(ev/'frontend-assets.json');dist=ev/'frontend-dist'
for base,expected,n in [(app,br['bundleFiles'],3),(dist,assets,7)]:
 assert len(expected)==n and {str(p.relative_to(base)) for p in base.rglob('*') if p.is_file()}==set(expected)
 for p,v in expected.items():assert sha((base/p).read_bytes())==v
assert br['binarySHA256']==sha(Path(br['binary']).read_bytes())=='a3ba0fd767e18033b27d2af1f03d20f1de1975bdfb4383d0b574a91376605281'
assert br['embeddedFiles']==assets and br['embeddedFrontendPath']==str(dist)
config=j(ev/'tauri-override.json');assert config==br['effectiveConfigOverride']==json.loads(br['buildEnvironment']['TAURI_CONFIG']);assert config['build']['frontendDist']==str(dist)
info=plistlib.loads((app/'Contents/Info.plist').read_bytes());assert info['CFBundleIdentifier']==config['identifier']=='com.rivune.desktop.qa.sentdraft20260910';assert info['CFBundleVersion']=='2026091005';profile=Path(info['LSEnvironment']['RIVUNE_ISOLATED_PROFILE_DIR']);assert profile==ev/'profile-reserved-not-launched' and not profile.exists() and not profile.is_symlink()
old=j(ev/'preserved-before.json');assert len(old)==522 and old==j(ev/'preserved-after.json')
for p,v in old.items():assert sha(Path(p).read_bytes())==v,p
# Recorded compiler dependency mapping only; no asset extraction from executable.
dep=(ev/'native-asset-dependencies.d').read_text().replace('\\ ', ' ')
for p in assets:assert str(dist/p) in dep,p
prior=j(r/'qa-artifacts/results-native-build-20260910/frontend-assets.json');same=[p for p,v in assets.items() if prior.get(p)==v];assert len(same)==5
entry=(dist/'desktop-entry.mjs').read_text();assert entry.startswith("import './desktop-host.mjs';\n") and 'index-CqRlVGz3.js' in entry and 'index-Hm6hc6-g.js' not in entry;assert './desktop-entry.mjs' in (dist/'index.html').read_text()
assert all(x['exitCode']==0 for x in br['commands']) and len(br['commands'])==2;assert 'Finished `dev` profile' in (ev/'native-build.log').read_text()
result={'verdict':'BOUNDED PASS','sourceCommit':c,'gitTree':tree,'aggregate':aggregate,'sourceFiles':148,'bundleFiles':3,'frontendAssets':7,'binarySHA256':br['binarySHA256'],'identifier':info['CFBundleIdentifier'],'reservedProfileAbsent':True,'preservedBeforeAfterAndCurrentFiles':522,'dependencyMappedFrontendInputs':7,'priorFrontendFilesUnchanged':same,'status':status,'embeddedAssetsExtracted':False,'nativeRuntimeTested':False}
(out/'result.json').write_text(json.dumps(result,indent=2)+'\n');print(json.dumps(result,indent=2))
